docs/guide/writing-migrations.md

```diff
@@ -0,0 +1,302 @@
+  extend: '@apostrophe/widget-type',
+  init(self) {
+    self.apos.migration.add('align-images', self.alignImages);
+  }
+  methods(self) {
+    return {
+      async alignImages(self) {
+        await self.apos.migration.eachWidget({},
+          async (doc, widget, dotPath) => {
+          if (widget.type !== '@apostrophecms/image') {
+            return;
+          }
+          if (widget.alignment === undefined) {
+            await self.apos.doc.db.updateOne({
+              _id: doc._id,
+            }, {
+              $set: {
+                [`${dotPath}.alignment`]: 'center'
+              }
+            });
+          }
+        });
+      }
+    }
+  }
+};
+```
+
+- The `eachWidget` method iterates over **every** widget in **every** area in **every** document. For this reason, you should check the `widget.type` to make sure you are only altering the desired widgets.
+- In our `criteria` argument we are passing an empty object, indicating that every document should be checked. You can narrow this focus if you only want the widgets on a certain document type changed. For example, passing `type: 'product'` would only change widgets that are in a product piece-type.
+- In the iterator, we first confirm that the widget is an image widget by checking the `widget.type`. If it is an image, we then check if the `alignment` property is present. If the `alignment` property is missing, we use `$set` to add it.
+
+The `iterator` in an `eachWidget` method gets three arguments. In addition to the document, `doc`, where the widget is found, it also receives the `widget` object that will be modified and the `dotPath`. The `dotPath` argument represents the location of the current widget within the document's structure, using a "dot notation" format. It allows you to trace exactly where the widget is nested within its parent area, such as `main.content.0`, where `main` is the area, `content` is the widget array, and `0` is the first widget in that array. This simplifies the process of pointing the MongoDB operation at the correct widget within a document.
+
+## Removing a Property from Existing Widgets
+
+Here’s how you can remove a property from widgets using `$unset`. Again, this is an irreversible operation, so you may want to simply remove a schema field. In this case, we are removing the `border` property from all `video` widgets:
+
+```js
+module.exports = {
+  extend: '@apostrophe/widget-type',
+  init(self) {
+    self.apos.migration.add('remove-vid-border', self.removeVidBorder);
+  }
+  methods(self) {
+    return {
+      async removeVidBorder(self) {
+        await self.apos.migration.eachWidget({},
+          async (doc, widget, dotpath) => {
+            if (widget.type !== '@apostrophecms/video') {
+              return;
+            }
+            if (widget.border !== undefined) {
+              await self.apos.doc.db.updateOne({
+                _id: doc._id
+              }, {
+                $unset: { `${dotPath.border}`: '' }
```

Comment:
Incorrect logic here, `.border` is supposed to be part of the string, not evaluated as a property of `dotPath` (you need to close your `}` sooner)

Reply:
Changed.